packages/json-rpc/polkadot-sdk-compat/src/parsed-enhancers/translate.ts

```diff
@@ -34,10 +34,16 @@ export const translate: ParsedJsonRpcEnhancer = (base) => {
       result,
     }: {
       id: string
-      result: { methods: string[] }
+      result: { methods: string[] } | undefined
     }) => {
       if (id !== RPC_METHODS_ID || !isRunning) return
 
+      // it's an error, let's try again
+      if (!result) {
+        setTimeout(sendMethodsRequest, 200)
+        return
+      }
+
```

Comment:
should we log the error when it happens? I'm afraid if this can get stuck here with no apparent reason.

Reply:
yeah, good point! Also, after a certain amount of retrials it should just propagate the error... 🤔

Reply:
Fixed!